xtask/src/release.rs

```diff
@@ -68,87 +65,92 @@ impl flags::Release {
-                    config.user = line[7..].trim().trim_matches('"').into();
-                }
-                l if l.starts_with("token =") => {
-                    config.token = line[8..].trim().trim_matches('"').into();
-                }
-                _ => {}
-            }
+    fn get_remote(&self) -> Result<String> {
+        let branch = cmd!("git rev-parse --abbrev-ref HEAD").read()?;
+        let remote = cmd!("git config branch.{branch}.remote").read()?;
 
-            if config.is_loaded() {
-                break;
-            }
+        if remote.is_empty() {
+            Err("Could not get current git remote")?
         }
 
-        if !config.is_loaded() {
-            Err("Some fields are missing for GitHub in the config file")?;
-        }
+        Ok(remote)
+    }
 
-        config.repo = {
-            let repo_url = cmd!("git remote get-url --push").arg(&config.remote).read()?;
-            repo_url
-                .as_str()
-                .trim_start_matches("git@github.com:")
-                .trim_start_matches("https://github.com/")
-                .trim_end_matches(".git")
-                .into()
-        };
+    /// Get the current version of the crate from the manifest.
+    fn get_version(&self) -> Result<String> {
+        let manifest_toml = read_file(project_root()?.join(&self.name).join("Cargo.toml"))?;
+        let manifest: CargoManifest = from_toml_str(&manifest_toml)?;
 
-        Ok(config)
+        Ok(manifest.package.version)
     }
 }
 
-/// The configuration used to make a release to GitHub.
-#[derive(Debug, Default)]
-struct GithubConfig {
-    /// The local name of the git remote to push to, from the config file.
-    remote: String,
-
-    /// The username to use for authentication, from the config file.
-    user: String,
+/// The required cargo manifest data of a crate.
+#[derive(Debug, Deserialize)]
+struct CargoManifest {
+    /// The package information.
+    package: CargoPackage,
+}
 
-    /// The personal access token to use for authentication, from the config file.
-    token: String,
+/// The required package information from a crate's cargo manifest.
+#[derive(Debug, Deserialize)]
+struct CargoPackage {
+    /// The package version.
+    version: String,
+}
 
-    /// The remote repository on GitHub as `owner/repo`.
-    repo: String,
+/// String manipulations for crate release.
+trait StrExt {
+    /// Remove soft line breaks.
+    fn trim_softbreaks(&self) -> String;
 }
 
-impl GithubConfig {
-    /// Create an empty config.
-    fn new() -> Self {
-        Default::default()
-    }
+impl StrExt for str {
+    fn trim_softbreaks(&self) -> String {
+        let mut string = "".to_owned();
+        let mut current = 0;
+        let mut chars = self.char_indices();
+
+        while let Some(p) = self[current..].find('\n') {
+            let pos = current + p;
+            string.push_str(&self[current..pos]);
+            // Keep hard line breaks (multiple `\n`s).
```

Comment:
I'm not sure we're going to need it, but there's also explicit line breaks in markdown, which are lines that end in (at least) two spaces. Those should be kept. And "hard line break" is not really the right term. Two newlines indicate a new paragraph, not a line break.

Reply:
Yeah actually I got terms mixed up. What you're talking about is called "hard line break" in the CommonMark spec. There's also `\` at the end of the line.